packages/thirdweb/src/react/web/ui/prebuilt/thirdweb/ClaimButton/index.tsx

```diff
@@ -87,12 +89,26 @@
     chain,
   });
   const account = useActiveAccount();
+  const { mutateAsync } = useSendAndConfirmTransaction();
   // TODO (pay): fetch nft metadata and set it as the payOptions metadata
   return (
     <TransactionButton
-      transaction={async () =>
-        await getClaimTransaction({ contract, account, claimParams })
-      }
+      transaction={async () => {
+        const claimTx = await getClaimTransaction({
+          contract,
+          account,
+          claimParams,
+        });
+        const approveTx = await getApprovalForTransaction({
+          transaction: claimTx,
+          // @ts-ignore We know it's not `undefined` fixme
```

Comment:
Why not just check if defined and throw or return early?

Reply:
you are right. throw should work. not return tho because it wants a preparedtransaction